web/home/src/pages/Home/home.tsx

```diff
@@ -67,82 +71,103 @@ function Home() {
     return res.data;
   };
 
-// Add the fake plugins
-
+  // Contains the list of plugins populated by fakeplugins
   const [plugins, setPlugins] = useState<PluginHomePage[]>(fakePluginsList);
-
-  // fetch plugins every 10 seconds
-  useEffect(() => {
-    // used to check if plugin list has changed
-    let previousFetch: PluginHomePage[] = [];
-    // fetch plugins
-    const interval = setInterval(() => {
-      getPlugins().then((res: PluginHomePage[]) => {
-        let combinedPlugins: PluginHomePage[] = [
-          ...fakePluginsList,
-          ...res,
-        ];
+  // Keep track of the previous fetch
+  const [previousFetch, setPreviousFetch] = useState<PluginHomePage[]>([]);
+  const displayPlugins = async () => {
+    await getPlugins()
+      .then((res: PluginHomePage[]) => {
+        let combinedPlugins: PluginHomePage[] = [...fakePluginsList, ...res];
         // If the list of plugins has changed, update the state
         if (JSON.stringify(previousFetch) !== JSON.stringify(res)) {
           setPlugins(combinedPlugins);
-          previousFetch = res;
+          setPreviousFetch(res);
         }
       })
       .catch((err) => {
         console.log(err);
         // If there is an error, use the previous list
-        setPlugins(previousFetch);
+        setPlugins((previousFetch) =>
+          previousFetch.length > 0 ? previousFetch : fakePluginsList,
+        );
       });
-
+  };
+  // fetch plugins every 10 seconds
```

Comment:
Please, let me know if you need help with that when you reach your scaling task card. 😇

Reply:
Yep for sure , i think we will try to setup a longpooling on the rebranding